File: 2020/day01/go/part2/part2.go
package part2

import "advent"
import "sort"
import "strconv"

func Parse(lines []string) []int {
	var values []int
	for _, line := range lines {
		value, err := strconv.Atoi(line)
		if err != nil {
			panic(err)
		}
		values = append(values, value)
	}
	return values
}

func FindTriple(values []int, target int) [3]int {
	distinct_values := advent.NewIntSet2(values)
	result := [3]int{0, 0, 0}

	for firstValue, _ := range distinct_values {
		for secondValue, _ := range distinct_values {
			if firstValue == secondValue {
				continue
			}
			thirdValue := target - firstValue - secondValue
			if (thirdValue != firstValue) && (thirdValue != secondValue) && (distinct_values.Has(thirdValue)) {
				result[0] = firstValue
				result[1] = secondValue
				result[2] = thirdValue
				sort.Ints(result[:])
				return result
			}
		}
	}
	return result
}

func ComputeAnswer(lines []string, target int) int {
	values := Parse(lines)
	results := FindTriple(values, target)
	return results[0] * results[1] * results[2]
}

File: 2015/day05/python/part2.py
from typing import List


def check1(line: str) -> bool:
    for index in range(len(line) - 3):
        # print(line, len(line), index, index+2, line[index:index+2])
        for index2 in range(index + 2, len(line) - 1):
            # print(line, len(line), index, index+2, index2, index2+2, line[index:index+2], line[index2:index2+2])
            if line[index : index + 2] == line[index2 : index2 + 2]:
                return True
    return False


def check2(line: str) -> bool:
    for index in range(len(line) - 2):
        if line[index] == line[index + 2]:
            return True
    return False


def is_nice(line: str) -> bool:
    return check1(line) and check2(line)


def compute_answer(lines: List[str]) -> int:
    result = 0
    for line in lines:
        if is_nice(line.lower()):
            result += 1
    return result
